### PR TITLE
FIX: changes Generic Motor exhaust velocity to cached property

### PR DESCRIPTION
**Pull request by giovaniceotto**
## Pull request type

- [x] Code changes (bugfix, features)

## Checklist

- [ ] Tests for the changes have been added (if needed)
- [x] Docs have been reviewed and added / updated
- [x] Lint (`black rocketpy/ tests/`) has passed locally 
- [x] All tests (`pytest --runslow`) have passed locally
- [x] `CHANGELOG.md` has been updated (if relevant)

## Current behavior

`GenericMotor.exhaust_velocity` is being decorated by `funcify_method`, which makes it (a float essentially) into a `lambda` defined `Function`. This causes everything that depends on it algebraically to become `lambda` defined `Functions` as well.

More specifically, this contaminates `GenericMotor.mass_flow_rate`, which in turn contaminates `GenericMotor.propellant_mass` (which actually includes an `integral_function` of the `lambda` `GenericMotor.exhaust_velocity` 🫣). Well, in the end, this contaminates `Rocket.total_mass`, which is used extensively by the `Flight` class during a simulation.

In summary, this use of `funcify_method` causes the `Flight` class to take forever (literally) to run.

## New behavior

What can I say? Changing the decorator from `funcify_method` to `cached_property` makes the sun shine once again, solving everything.

## Breaking change

- [x] No

### CODE REVIEW

**Review by phmbressan — DISMISSED** [review]
Thank you for going through the journey to discover the bottleneck in this class. I was postponning for some time. Great finds!

Won't even start about snowball effect that the `Function` is currently generating that slows down the code so much. I will only say that the sooner this is fixed, the better.

**Review by Gui-FernandesBR — APPROVED** [review]
I swear that I tried but I honestly have no idea of how you discovered this error in such a chain of calls.

**Comment by Gui-FernandesBR** [comment]
Before merging, do you see any possibility for new unit tests here? It's quite complicated because the error was only happening due to a "snowball" effect.

**Review by Gui-FernandesBR — APPROVED** [review]
LGTM

File: rocketpy/motors/motor.py [hunk 1]
```diff
@@ -1224,7 +1224,9 @@ def exhaust_velocity(self):
         self.exhaust_velocity : Function
             Gas exhaust velocity of the motor.
         """
-        return self.total_impulse / self.propellant_initial_mass
+        return Function(
+            self.total_impulse / self.propellant_initial_mass
+        ).set_discrete_based_on_model(self.thrust)
```
Comment (MateusStano) [inline rocketpy/motors/motor.py:1229]: Do we need to return a Function here if this is already a `@funcify_method` ?

**Review by MateusStano — APPROVED** [review]

File: rocketpy/motors/motor.py (hunk 1 above)
Comment (phmbressan) [reply rocketpy/motors/motor.py:1229]: The Function is called so we can discretize it. If we applied @funcify directly, the Function source would be a numerical value.

The main issue this PR tackles are hidden lambda based Functions that end up cascading the others in arithmetic, making code slow.
